pkgs/test_core/lib/src/runner/debugger.dart

```diff
@@ -85,7 +85,7 @@ class _Debugger {
   /// Whether [close] has been called.
   bool _closed = false;
 
-  bool get _json => _config.reporter == 'json';
```

Comment:
I think it only care about whether it is allowed to write to `stdout`, so we probably don't need to worry about whether json is used as a file reporter.

Reply:
Fixed, although I left the other check related to the json reporter just below this as I think collecting traces would need to happen when the json file reporter is used, too.